rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +603,223 @@ def _query_lending_balances(
+        - BinancePermissionError
+        - RemoteError
+        """  # noqa: E501
+        data = []
+        current_page = 1
+        while True:
+            if additional_options:
+                additional_options.update({'current': current_page})
+
+            response = self._api_query_list_within_time_delta(
+                api_type=api_type,
+                start_ts=start_ts,
+                end_ts=end_ts,
+                time_delta=time_delta,
+                method=method,
+                additional_options=additional_options,
+            )
+            if len(response) == 0:
+                break
+
+            data.extend(response)
+            current_page += 1
+
+        return data
+
+    def _query_lending_interests_history(
+            self,
+            cursor: 'DBCursor',
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> None:
+        """Queries Binance lending interest history, transforms it into `HistoryBaseEntry` objects and save it in the database.
+
+        "DAILY" -> flexible savings.
+        "CUSTOMIZED_FIXED" -> fixed savings.
+        "ACTIVITY" -> fixed savings but as a result of token partners.
+
+        `start_ts` - `end_ts` <= 30 days.
+        `size` is set as 100 to reduce network calls, and it's the maximum number that can be selected.
+
+        Lending Interest History Documentation:
+        https://binance-docs.github.io/apidocs/spot/en/#get-interest-history-user_data-2
+
+        May raise:
+        - RemoteError
+        - BinancePermissionError
+        """  # noqa: E501
+        ranges = DBQueryRanges(self.db)
+        range_query_name = f'{self.location}_lending_history_{self.name}'
+        ranges_to_query = ranges.get_location_query_ranges(
+            cursor=cursor,
+            location_string=range_query_name,
+            start_ts=start_ts,
+            end_ts=end_ts,
+        )
+        for query_start_ts, query_end_ts in ranges_to_query:
+            events = []
+            for lending_type in ('DAILY', 'ACTIVITY', 'CUSTOMIZED_FIXED'):
+                response = self._api_query_list_paginated_within_time_delta(
+                    api_type='sapi',
+                    start_ts=query_start_ts,
+                    time_delta=BINANCE_LENDING_INTEREST_HISTORY_INTERVAL_TS,
+                    end_ts=query_end_ts,
+                    method='lending/union/interestHistory',
+                    additional_options={'lendingType': lending_type, 'size': 100},
+                )
+                for entry in response:
+                    try:
+                        interest_received = deserialize_asset_amount_force_positive(entry['interest'])  # noqa: E501
+                        if interest_received == ZERO:
+                            continue
+
+                        event_identifier = HistoryBaseEntry.deserialize_event_identifier(str(uuid4()))  # noqa: E501
+                        timestamp = deserialize_timestamp_ms(entry['time'])
```

Comment:
But this change is not doing the same as before. Before you had a conversion using exact true and now you are just taking the same value. Is this what you wanted?

Reply:
Yeah, it was a silly copy pasta.
Binance returns a timestamp in milliseconds.